### PR TITLE
Update Deployment spec on ContainerSource change

### PR DESCRIPTION
**Pull request by grantr**
Without this the ContainerSource never reconciles the Deployment. Tests will be coming soon in a different PR.

## Proposed Changes

  * Update Deployment owned by a ContainerSource when the ContainerSource spec changes.

**Release Note**

```release-note
ContainerSource updates owned Deployment when spec changes.
```

/cc @n3wscott @vaikas-google

### CODE REVIEW

File: pkg/controller/containersource/reconcile.go [hunk 1]
```diff
@@ -98,15 +99,26 @@ func (r *reconciler) Reconcile(ctx context.Context, object runtime.Object) (runt
 			}
 			r.recorder.Eventf(source, corev1.EventTypeNormal, "Deployed", "Created deployment %q", deploy.Name)
 			source.Status.MarkDeploying("Deploying", "Created deployment %s", args.Name)
-		} else {
-			return source, err
+			// Wait for the deployment to get a status
```
Comment (vaikas-google) [inline pkg/controller/containersource/reconcile.go]: I don't understand what this comment means, can you clarify.

Comment (n3wscott) [reply pkg/controller/containersource/reconcile.go]: I think it means wait for the deployment to be reconciled at least once.

**Comment by n3wscott** [comment]
/lgtm 
/hold
unhold when Ville is happy and questions are answered.

File: pkg/controller/containersource/reconcile.go (hunk 1 above)
Comment (grantr) [reply pkg/controller/containersource/reconcile.go]: Expanded, PTAL.

**Pull request merged**